src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_structure.py

```diff
@@ -151,30 +134,19 @@ def result(self, result: Result) -> None:
         self._result = result
 
     @staticmethod
-    def create_genotyping(data: pd.Series[str]) -> tuple[WellItem, WellItem]:
-        identifier = try_int_from_series(data, "Well")
-
-        snp_name = try_str_from_series(
-            data,
-            "SNP Assay Name",
+    def create_genotyping(data: SeriesData) -> tuple[WellItem, WellItem]:
+        identifier = data[int, "Well"]
+        snp_name = assert_not_none(
```

Comment:
nit: feel free to disagree if this is too much abstraction but it feels like `assert_not_none` is called a lot with `data.get`, maybe we could add it as a parameter to `.get` or add a new function?

Reply:
So - the issue here is that if you want to assert, you should just use the index operator.

But - these particular cases want to pass a custom error message. That said, I realized I can still support the custom error message in the index operator, I was just too tired to think of it last night 😓